File: python/docmost/services/ai_api.py
"""Generated API client."""
from __future__ import annotations

from typing import Any, Optional

import httpx

from docmost.http_client import DocmostHttpClient
from docmost.types import ApiResponse
from docmost.models import SearchDto


class AIApi:
    """AI API."""

    def __init__(self, http: DocmostHttpClient) -> None:
        self._http = http

    def ai_answers(self, request: SearchDto) -> ApiResponse[Any]:
        return self._http.post("ai/answers", body=request.model_dump(by_alias=True, exclude_none=True))

File: go/docmost/services/ai_api.go
package services

import (
	"context"

	"github.com/ggpo1/docmost-sdk/go/internal/httpclient"
	"github.com/ggpo1/docmost-sdk/go/docmost/models"
)

// AIAPI implements AI API.
type AIAPI struct {
	http *httpclient.HTTPClient
}

func NewAIAPI(http *httpclient.HTTPClient) *AIAPI {
	return &AIAPI{http: http}
}

func (a *AIAPI) AiAnswers(ctx context.Context, request *models.SearchDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "ai/answers", request)
}

